src/protocols/rdp/user.c

```diff
@@ -121,13 +121,16 @@ int guac_rdp_user_join_handler(guac_user* user, int argc, char** argv) {
 
         /* Inbound arbitrary named pipes */
         user->pipe_handler = guac_rdp_pipe_svc_pipe_handler;
-        
+
         /* If we own it, register handler for updating parameters during connection. */
         if (user->owner)
             user->argv_handler = guac_argv_handler;
 
     }
 
+    /* Process some resources when users leaving */
+    user->leave_handler = guac_rdp_user_leave_handler;
```

Comment:
If all users will always have the same handler, this can be easier/simpler when set at the `client` level. For example, in the VNC support:

https://github.com/apache/guacamole-server/blob/44145f681aa3cf08e9cf01e99a68096c77bcab39/src/protocols/vnc/client.c#L62

Reply:
`guac_rdp_user_leave_handler()` performs the removing of common cursor and the free of `guac_rdp_settings` allocated for each shared user.
But the free to `guac_rdp_settings` of owner is performed through guac_rdp_client_free_handler().
Please refer [this link](https://issues.apache.org/jira/browse/GUACAMOLE-1350?focusedCommentId=17350114&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-17350114) for more information.
So, I think that we can't set this handler at `client` level.